src/driver/amdxdna/ve2_debug.c

```diff
@@ -0,0 +1,359 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Copyright (C) 2025, Advanced Micro Devices, Inc.
+ */
+#include <linux/device.h>
+#include <linux/version.h>
+
+#include "ve2_fw.h"
+#include "ve2_mgmt.h"
+#include "ve2_of.h"
+
+static int ve2_get_hwctx_status(struct amdxdna_client *client, struct amdxdna_drm_get_info *args)
+{
+	size_t hwctx_data_sz = sizeof(struct amdxdna_drm_query_ctx);
+	struct amdxdna_drm_query_ctx *hwctx_data;
+	struct amdxdna_drm_query_ctx __user *buf;
+	struct amdxdna_dev *xdna = client->xdna;
+	struct amdxdna_client *tmp_client;
+	u32 req_bytes = 0, hw_i = 0;
+	struct amdxdna_ctx *hwctx;
+	unsigned long hwctx_id;
+	bool overflow = false;
+	int ret = 0, idx;
+
+	hwctx_data = kzalloc(hwctx_data_sz, GFP_KERNEL);
+	if (!hwctx_data)
+		return -ENOMEM;
+
+	buf = u64_to_user_ptr(args->buffer);
+
+	list_for_each_entry(tmp_client, &xdna->client_list, node) {
+		idx = srcu_read_lock(&tmp_client->ctx_srcu);
+		amdxdna_for_each_ctx(tmp_client, hwctx_id, hwctx) {
+			req_bytes += hwctx_data_sz;
+			if (args->buffer_size < req_bytes) {
+				/* Continue iterating to get the required size */
+				overflow = true;
```

Comment:
why overflow is required? checking if (args->buffer_size < req_bytes) should be good enough?